sound/soc/sof/topology.c

```diff
@@ -146,11 +148,19 @@ static int sof_keyword_dapm_event(struct snd_soc_dapm_widget *w,
 	dev_dbg(sdev->dev, "received event %d for widget %s\n",
 		event, w->name);
 
+	/* get runtime PCM params using widget's stream name */
+	spcm = snd_sof_find_spcm_name(sdev, swidget->widget->sname);
+
 	/* process events */
 	switch (event) {
 	case SND_SOC_DAPM_PRE_PMU:
+		if (spcm->stream[stream].suspend_ignored) {
```

Comment:
@plbossart let me try to explain the flow in the order in which they happen. @keyonjie correct me if I am wrong:
During suspend: 
1. system suspend prepare callback -> sets S0_suspend flags
2. Streams are suspended with a TRIGGER_SUSPEND -> sets the suspend_ignored flag
3. DAPM widgets are powered down -> POST_PMD event -> if suspend_ignored flag set-> do nothing. else stop the pipeline
4. SOF device suspend callback 

During system resume
1. SOF device resume callback
2. system resume complete callback -> sets the s0_suspend flag to false.
3. DAPM widgets powered up -> check if suspend_ignored and do nothing, else start the pipeline
4. TRIGGER_START for all pipelines -> unset the suspend_ignored flag

Does this make sense?

Reply:
yes, but we don't have a case with TRIGGER_RESUME, so it should be marked with some sort of warning saying the code can't really work.

Reply:
@plbossart do you mean adding some comments in the trigger_resume case, e.g.:
```
       case SNDRV_PCM_TRIGGER_RESUME:
		/*
                 * Only happens when INFO_RESUME is supported, 
                 * no need to resume streams that were not suspended.
                 */
...
		/* fallthrough */
	case SNDRV_PCM_TRIGGER_START:
		/*
                 * This may arrive at stream resuming where INFO_RESUME is
                 * not supported, no need to re-start it if the stream was
                 * not suspended in this case.
                 */
...
```

Reply:
yes, maybe reworded a bit

````
case SNDRV_PCM_TRIGGER_RESUME:
		/*
                 * this case will be triggered when INFO_RESUME is supported, 
                 * no need to resume streams that remained enabled in D0ix.
                 */
...
		/* fallthrough */
	case SNDRV_PCM_TRIGGER_START:
		/*
                 * This case will be triggered when INFO_RESUME is not supported
                 * no need to re-start streams that remained enabled in D0ix.
                 */

````